runtime/testy/runtime.py

```diff
@@ -66,16 +73,23 @@ def runtime():
 def runStudentCode(badThingsQueue, stateQueue, pipe, testName = "", maxIter = 0):
   try:
     import signal
+
+    terminated = False
+    def sigTermHandler(signum, frame):
+      global terminated
```

Comment:
Shouldn't this be nonlocal as we only want to change it in the parent frame and not global